data/org.eclipse.birt.report.data.oda.jdbc.ui/src/org/eclipse/birt/report/data/oda/jdbc/ui/editors/SQLDataSetEditorPage.java

```diff
@@ -1075,4 +1119,149 @@ private void insertTreeItemText() {
+				.setText(JdbcPlugin.getResourceString("tablepage.querytext.find.option.case.sensitive")); //$NON-NLS-1$
+		this.findQueryTextCaseSensitive
+				.setToolTipText(JdbcPlugin.getResourceString("tablepage.querytext.find.option.case.sensitive.tooltip")); //$NON-NLS-1$
+		this.findQueryTextCaseSensitive.setSelection(false);
+		this.findQueryTextCaseSensitive.setLayoutData(csLayoutData);
+		this.findQueryTextCaseSensitive.setEnabled(true);
+
+		GridData wwLayoutData = new GridData(GridData.HORIZONTAL_ALIGN_BEGINNING);
+		this.findQueryTextWholeWord = new Button(group, SWT.CHECK);
+		this.findQueryTextWholeWord.setText(JdbcPlugin.getResourceString("tablepage.querytext.find.option.whole.word")); //$NON-NLS-1$
+		this.findQueryTextWholeWord
+				.setToolTipText(JdbcPlugin.getResourceString("tablepage.querytext.find.option.whole.word.tooltip")); //$NON-NLS-1$
+		this.findQueryTextWholeWord.setSelection(false);
+		this.findQueryTextWholeWord.setLayoutData(wwLayoutData);
+		this.findQueryTextWholeWord.setEnabled(true);
+	}
+
+	private void setupFindQueryTextButtons(Group group) {
+
+		GridData fwButtonFindTextLayoutData = new GridData(GridData.HORIZONTAL_ALIGN_BEGINNING);
+		Button fwButtonFindText = new Button(group, SWT.BUTTON1);
+		fwButtonFindText.setText("▼ " + JdbcPlugin.getResourceString("tablepage.querytext.find.button.forward")); //$NON-NLS-1$
+		fwButtonFindText.setToolTipText(JdbcPlugin.getResourceString("tablepage.querytext.find.button.forward.tooltip")); //$NON-NLS-1$
+		fwButtonFindText.setLayoutData(fwButtonFindTextLayoutData);
+		fwButtonFindText.setEnabled(true);
+		// Add listener to the find query button "forward"
+		fwButtonFindText.addSelectionListener(new SelectionAdapter() {
+
+			@Override
+			public void widgetSelected(SelectionEvent event) {
+				PlatformUI.getWorkbench().getDisplay().asyncExec(new Runnable() {
+
+					@Override
+					public void run() {
+						findQueryTextForward();
+					}
+				});
+			}
+		});
+
+		GridData bwButtonFindTextLayoutData = new GridData(GridData.HORIZONTAL_ALIGN_BEGINNING);
+		Button bwButtonFindText = new Button(group, SWT.BUTTON1);
+		bwButtonFindText.setText("▲ " + JdbcPlugin.getResourceString("tablepage.querytext.find.button.backward")); //$NON-NLS-1$
+		bwButtonFindText.setToolTipText(JdbcPlugin.getResourceString("tablepage.querytext.find.button.backward.tooltip")); //$NON-NLS-1$
+		bwButtonFindText.setLayoutData(bwButtonFindTextLayoutData);
+		bwButtonFindText.setEnabled(true);
+		// Add listener to the find query button "forward"
+		bwButtonFindText.addSelectionListener(new SelectionAdapter() {
+
+			@Override
+			public void widgetSelected(SelectionEvent event) {
+				PlatformUI.getWorkbench().getDisplay().asyncExec(new Runnable() {
+
+					@Override
+					public void run() {
+						findQueryTextBackward();
+					}
+				});
+			}
+		});
+	}
+
+	private void setupLabelFindQueryTextGroup(int findMode) {
+		Device device = Display.getCurrent();
+		Color fontColor = new Color(device, 0, 0, 0);
```

Comment:
Assuming you’re not in dark mode?

Reply:
When the color of the label is changed to "red" for the hint message it will be set back to default-color black.
Therefore I define the default color. We need the color-reset because if not the label would be always red when a red-message was displayed.

Reply:
What color is the text in dark mode? I’m on a phone so it limits the links to send.  Setting to black assumes the text is black. Not a good assumption. Also Color needs to be disposed. Better to look how other parts of the framework get colors and set them.

Reply:
Done and optimized.